Gemfile

```diff
@@ -99,12 +66,127 @@ gem "ruport",                         "=1.7.0",                       :git => "h
 # https://github.com/jeremyevans/ruby-american_date
 gem "american_date"
 
-group :ui_dependencies do # Added to Bundler.require in config/application.rb
+# Make sure to tag your new bundler group with the manageiq_default group in addition to your specific bundler group name.
+# This default is used to automatically require all of our gems in processes that don't specify which bundler groups they want.
+#
+### providers
+group :amazon, :manageiq_default do
+  manageiq_plugin "manageiq-providers-amazon"
+end
+
+group :ansible, :manageiq_default do
+  gem "ansible_tower_client",           "~>0.12.2",      :require => false
+end
+
+group :azure, :manageiq_default do
+  manageiq_plugin "manageiq-providers-azure"
+end
+
+group :foreman, :manageiq_default do
+  manageiq_plugin "manageiq-providers-foreman"
+  gem "foreman_api_client",             ">=0.1.0",   :require => false, :git => "https://github.com/ManageIQ/foreman_api_client.git", :branch => "master"
+end
+
+group :google, :manageiq_default do
+  manageiq_plugin "manageiq-providers-google"
+  gem "fog-google",                     ">=0.5.2",       :require => false
+  gem "google-api-client",              "~>0.8.6",       :require => false
+end
+
+group :hawkular, :manageiq_default do
+  manageiq_plugin "manageiq-providers-hawkular"
+end
+
+group :kubernetes, :openshift, :manageiq_default do
+  manageiq_plugin "manageiq-providers-kubernetes"
+end
+
+group :lenovo, :manageiq_default do
+  manageiq_plugin "manageiq-providers-lenovo"
+end
+
+group :nuage, :manageiq_default do
+  manageiq_plugin "manageiq-providers-nuage"
+end
+
+group :openshift, :manageiq_default do
+  manageiq_plugin "manageiq-providers-openshift"
+  gem "htauth",                         "2.0.0",         :require => false # used by container deployment
+end
+
+group :openstack, :manageiq_default do
+  manageiq_plugin "manageiq-providers-openstack"
+end
+
+group :ovirt, :manageiq_default do
+  manageiq_plugin "manageiq-providers-ovirt"
+  gem "ovirt-engine-sdk",               "~>4.1.4",       :require => false # Required by the oVirt provider
+  gem "ovirt_metrics",                  "~>1.4.1",       :require => false
+end
+
+group :scvmm, :manageiq_default do
+  manageiq_plugin "manageiq-providers-scvmm"
+end
+
+group :vmware, :manageiq_default do
+  manageiq_plugin "manageiq-providers-vmware"
+  gem "vmware_web_service",             "~>0.1.1",       :require => false
+end
+
+### shared dependencies
+group :google, :openshift, :manageiq_default do
+  gem "sshkey",                         "~>1.8.0",       :require => false
```

Comment:
Yes, for reference, the goal was to create groups and move dependencies into the groups to make it obvious where things belong.  The next step is to push these dependencies to the correct place.  Putting these in groups makes this very obvious.